### PR TITLE
Add unit test for Feature/JWT Authorization 

### PR DESCRIPTION
**Pull request by dinhl**
closes https://github.com/hwsc-org/hwsc-file-transaction-svc/issues/43

Capture

### CODE REVIEW

File: auth/test_jwt.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+                         [
+                            ("eyJBbGciOjIsIlRva2VuVHlwIjoxfQ.eyJVVUlEIjoiMDFkM3gzd20ybm5yZGZ6cDB0a2Eydnc5ZHgiLCJQZXJtaXNzaW9uIjozLCJFeHBpcmF0aW9uVGltZXN0YW1wIjoxODkzNDU2MDAwfQ.8lVhZo_W6KmGI2oi5JNHioDvPq2Yl86v4uae3RfKc-qoKUwHNxFtXO2NFmChsi35__t1uC_SD-Ay_MoateeDNg==" , does_not_raise()),
+                            ("eyJBbGciOjIsIlRva2VuVHlwIjoxfQ.eyJVVUlEIjoiMDFkM3gzd20ybm5yZGZ6cDB0a2Eydnc5ZHgiLCJQZXJtaXNzaW9uIjozLCJFeHBpcmF0aW9uVGltZXN0YW1wIjoxODkzNDU2MDAwfQ" , pytest.raises(ValueError)),
+                            ("eyJBbGciOjIsIlRva2VuVHlwIjoxfQ", pytest.raises(ValueError)),
+                            ("", pytest.raises(ValueError)),
+                        ])
+
+def test_validate_token_len(input, expectation):
+    with expectation:
+        jwt.validate_token_len(input)
```
Comment (dinhl) [inline auth/test_jwt.py]: i will an enter here in my code

File: auth/test_jwt.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import pytest
+from auth import jwt
+from contextlib import contextmanager
+
+
+@contextmanager
+def does_not_raise():
+    yield
```
Comment (faraonc) [inline auth/test_jwt.py]: FYI, you do not need this anymore. https://docs.pytest.org/en/latest/example/parametrize.html#parametrizing-conditional-raising

File: auth/test_jwt.py [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+                             ({'Alg': -1, 'TokenTyp': 1}, pytest.raises(ValueError)),
+                             ({'Alg': 3, 'TokenTyp': 1}, pytest.raises(ValueError)),
+                             ({'Alg': 1, 'TokenTyp': -1}, pytest.raises(ValueError)),
+                             ({'Alg': 1, 'TokenTyp': 3}, pytest.raises(ValueError)),
+                             ({'Alg': 0, 'TokenTyp': 1}, does_not_raise()),
+                             ({'Alg': 1, 'TokenTyp': 1}, does_not_raise()),
+                             ({'Alg': 2, 'TokenTyp': 1}, does_not_raise()),
+                             ({'Alg': 1, 'TokenTyp': 0}, does_not_raise()),
+                             ({'Alg': 1, 'TokenTyp': 1}, does_not_raise()),
+                             ({'Alg': 1, 'TokenTyp': 2}, does_not_raise()),
```
Comment (faraonc) [inline auth/test_jwt.py]: For all these ints, we should use enums. Hardcoding the int will break the tests if a new Alg or TokenTyp has been added

File: auth/test_jwt.py [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+    actual_output = jwt.get_decoded_header(input_str)
+    assert actual_output == expect_output
+
+
+@pytest.mark.parametrize("input_str, expect_output, description",
+                         [
+                            ("eyJVVUlEIjoiMDFkM3gzd20ybm5yZGZ6cDB0a2Eydnc5ZHgiLCJQZXJtaXNzaW9uIjozLCJFeHBpcmF0aW9uVGltZXN0YW1wIjoxODkzNDU2MDAwfQ", {'UUID': '01d3x3wm2nnrdfzp0tka2vw9dx', 'Permission': jwt.PermissionEnum.ADMIN, 'ExpirationTimestamp': 1893456000}, "test get decoded body"),
+                         ]
+                         )
+def test_get_decoed_body(input_str, expect_output, description):
```
Comment (faraonc) [inline auth/test_jwt.py]: typo on func name

**Comment by dinhl** [comment]
Capture

File: auth/jwt.py [hunk 5]
```diff
@@ -33,14 +32,25 @@ class PermissionEnum(IntEnum):
...
     if not token_str:
         raise ValueError("token string is None")
+
     result = token_str.strip().split('.')
     if len(result) != 3:
         raise ValueError("invalid token")
 
 
 def base64_decode(input_str):
+    """Decoded a token string
```
Comment (faraonc) [inline auth/jwt.py]: Decode

Comment (faraonc) [inline auth/jwt.py]: `Decode`

File: auth/jwt.py [hunk 6]
```diff
@@ -118,22 +158,27 @@ def validate_signature(header_body_token, secret_key, header_dict, signature_tok
...
+    token_list = token_str.strip().split('.')
+    # get the header.body token from the token_list
+    header_body_token = token_list[0] + "." + token_list[1]
+    # get the signature of the token from the token_list
+    signature_token = token_list[2]
+    # validate if the token_str has 3 elements
+    validate_token_len(token_str)
+    # decoded header token to get the header dictionary
+    header_dict = get_decoded_header(token_list[0])
+    # decoded body token to get the body dictionary
```
Comment (faraonc) [inline auth/jwt.py]: decode

File: auth/jwt.py [hunk 7]
```diff
@@ -118,22 +158,27 @@ def validate_signature(header_body_token, secret_key, header_dict, signature_tok
...
+
+    # split the token_str to get a list of header, body, signature
+    token_list = token_str.strip().split('.')
+    # get the header.body token from the token_list
+    header_body_token = token_list[0] + "." + token_list[1]
+    # get the signature of the token from the token_list
+    signature_token = token_list[2]
+    # validate if the token_str has 3 elements
+    validate_token_len(token_str)
+    # decoded header token to get the header dictionary
```
Comment (faraonc) [inline auth/jwt.py]: decode

**Pull request merged**